core/enumerator/product_spec.rb

```diff
@@ -2,6 +2,29 @@
 
 ruby_version_is "3.2" do
   describe "Enumerator.product" do
+    it "creates a finite enumerator" do
+      enum = Enumerator.product(1..2, ["A", "B"])
+      enum.to_a.should == [[1, "A"], [1, "B"], [2, "A"], [2, "B"]]
+    end
+
+    it "accepts a list of enumerators of any length" do
+      enum = Enumerator.product(1.., ["A"], ["B"])
+      enum2 = Enumerator.product(2.., ["A"], ["B"], ["C"])
+
+      enum.take(2).should == [[1, "A", "B"], [2, "A", "B"]]
+      enum2.take(2).should == [[2, "A", "B", "C"], [3, "A", "B", "C"]]
+    end
+
+    it "creates empty array with 0 arguments" do
+      enum = Enumerator.product
+      enum.to_a.should == [[]]
+    end
+
+    it "is an instance of Enumerator::Product" do
+      enum = Enumerator.product
+      enum.class.should == Enumerator::Product
+    end
+
```

Comment:
**minor:**
regarding the test case below:

```ruby
    it "creates an infinite enumerator" do
      enum = Enumerator.product(1.., ["A", "B"])
      enum.take(5).should == [[1, "A"], [1, "B"], [2, "A"], [2, "B"], [3, "A"]]
    end
```

I would reword the title and also check `#size` of the returned enumerator:

```ruby
    it "accepts infinite enumerators and returns infinite enumerator" do
      enum = Enumerator.product(1.., ["A", "B"])
      enum.take(5).should == [[1, "A"], [1, "B"], [2, "A"], [2, "B"], [3, "A"]]
      enum.size.should == Float::INFINITY
    end
```